draft-reddy-add-resolver-info-06.txt

```diff
@@ -0,0 +1,448 @@
+   small amount of useful information about a resolver.  The general
+   format rules for DNS TXT records are discussed in Section 6.1 of
+   [RFC6763] and the recommendations to limit the TXT record size are
+   discussed in Section 6.2 of [RFC6763].  The RESINFO RRtype uses key/
+   value pairs to convey the resolver information.  Each key/value pair
+   is encoded using the format rules in Section 6.3 of DNS-SD [RFC6763].
+   Using standardized key/value syntax within the RESINFO RRtype makes
+   it easier for additional keys to be defined later.  If an
+   implementation sees unknown keys in a RESINFO RRtype, it MUST
+   silently ignore them.  The rules for the keys is defined in
+   Section 6.4 of [RFC6763].
+
+   All keys MUST either be defined in the IANA registry Section 7.2 or
+   begin with the substring "temp-" for names defined for local use
+   only.
+
+5.  Resolver Information
+
+   The resolver information includes the following keys and values:
+
+   qnamemin:  If the DNS server supports QNAME minimisation [RFC9156] to
+
+
+
+Reddy & Boucadair         Expires 3 March 2023                  [Page 3]
+
+Internet-Draft          DNS Resolver Information             August 2022
+
+
+      improve DNS privacy, the key is present.  Note that, as per the
+      rules for the keys is defined in Section 6.4 of [RFC6763], if
+      there is no '=' in a key, then it is a boolean attribute, simply
+      identified as being present, with no value.  This is a mandatory
+      attribute.
+
+   exterr:  If the DNS server supports extended DNS error (EDE)
+      [RFC8914] to return additional information about the cause of DNS
+      errors, the parameter lists the possible extended DNS error codes
+      that can be returned by the DNS server.  When multiple values are
+      present, they are seperated by comma.  This is an optional
+      attribute.
+
+   infourl:  An URL that points to the generic unstructured resolver
+      information (e.g., DoH APIs supported, possible HTTP status codes
+      returned by the DoH server, how to report a problem) for
+      troubleshooting purpose.  The server MUST support the content-type
+      'text/html'.  The DNS client MUST reject the URL if the scheme is
+      not "https".  The client MUST validate that both the encrypted DNS
+      server and the resolver information server are owned and managed
+      by the same entity by establishing a TLS connection to the domain
+      name in the URL and checking if the subjectAltName entry in the
+      server certificate includes the name of the encrypted DNS server.
+      If this match fails, the client MUST ignore the resolver
+      information.  As such, the URL should be treated only as
+      diagnostic information for IT staff.  This is a mandatory
+      attribute.
+
+   New keys can be defined as per the procedure defined in Section 7.2.
+
+   Figure 1 shows an example of resolver information.
```

Comment:
I suggest we transform this example into a record as will be published in DNS.

Reply:
Good point, fixed.